packages/scheduler/src/Scheduler.js

```diff
@@ -474,17 +474,42 @@ if (
   typeof window === 'undefined' ||
   typeof window.addEventListener !== 'function'
 ) {
-  // If this accidentally gets imported in a non-browser environment, fallback
-  // to a naive implementation.
-  var timeoutID = -1;
-  requestHostCallback = function(callback, absoluteTimeout) {
-    timeoutID = setTimeout(callback, 0, true);
+  // If Scheduler runs in a non-DOM environment, it falls back to a naive
+  // implementation using setTimeout. This only meant to be used for testing
+  // purposes, like with jest's fake timer API.
+  var _callback = null;
+  var _currentTime = -1;
+  function flushCallback(didTimeout, ms) {
+    if (_callback !== null) {
+      var cb = _callback;
+      _callback = null;
+      try {
+        _currentTime = ms;
+        cb(didTimeout);
+      } finally {
+        _currentTime = -1;
+      }
+    }
+  }
+  requestHostCallback = function(cb, ms) {
+    if (_currentTime !== -1) {
+      // Protect against re-entrancy. Jest's fake timer queue flushes timer
```

Comment:
Maybe remove mentions of Jest? If it’s meant to be generic.

Reply:
Good point